addons/mail/static/src/components/composer_text_input/composer_text_input.js

```diff
@@ -341,13 +364,28 @@ Object.assign(ComposerTextInput, {
     components,
     defaultProps: {
         hasMentionSuggestionsBelowPosition: false,
-        hasSendOnEnterEnabled: true,
+        sendShortcuts: [],
     },
     props: {
         composerLocalId: String,
         hasMentionSuggestionsBelowPosition: Boolean,
-        hasSendOnEnterEnabled: Boolean,
         isCompact: Boolean,
+        /**
+         * Keyboard shortcuts from text input to send message.
+         */
+        sendShortcuts: {
+            type: Array,
+            element: String,
+            validate: prop => {
+                for (const shortcut of prop) {
+                    if (!['ctrl-enter', 'enter', 'meta-enter'].includes(shortcut)) {
+                        return false;
+                    }
+                }
+                return true;
+            },
```

Comment:
I have no idea but isn't it possible to define "shape" or similar on array, instead of having to manually iterate elements to validate them? I just feels not natural/too much code, but if not possible then it's ok.

Reply:
Would feel more natural if props validation could define `validate` function in the scope of `element`, like:
```js
element: {
   type: String,
   validate: val => [/* authorized vals */].includes(val),
},
```
But `validate` only works on prop itself.

Reply:
Ok. That would be a good issue to open on OWL repo I think.